packages/plugin-react/src/index.ts

```diff
@@ -34,12 +34,6 @@ export interface Options {
    * @default "react"
    */
   jsxImportSource?: string
-  /**
-   * Set this to `true` to annotate the JSX factory with `\/* @__PURE__ *\/`.
-   * This option is ignored when `jsxRuntime` is not `"automatic"`.
-   * @default true
-   */
-  jsxPure?: boolean
```

Comment:
esbuild assumes JSX is always side effect-free. I'm doubtful this option is ever actually used because `jsxPure: true` was set as the default option.

An alternative would be to keep the existing babel-based transformation only if `jsxPure: false`, but I'm not sure there's actually a use case for this.

Reply:
Yes, I think we can remove it and issue a breaking change warning if it is false in the next major.

Reply:
The PR wanted it to force it to be `true`, no? I think we should remove the option and always make it `true` as it is useful as you quoted

Reply:
Ah yes, good point!

Reply:
`pure` of `ESBuildOptions` is `string[]` (https://esbuild.github.io/api/#pure), which only can add.

Here seems no way to disable pure annotations of jsx-runtimes.

Reply:
> esbuild assumes JSX is always side effect-free. I'm doubtful this option is ever actually used because `jsxPure: true` was set as the default option.
> 
> An alternative would be to keep the existing babel-based transformation only if `jsxPure: false`, but I'm not sure there's actually a use case for this.

Heads up that there is now a use case for this. Someone created a use case for JSX syntax that requires free-standing JSX elements as top-level statements instead of using `return`: https://github.com/vuoro/rahti. See also: https://github.com/evanw/esbuild/issues/2539. I will be adding a setting to esbuild for this.

Reply:
I've updated this PR to use the new JSX side effects option in esbuild